include/ada/url-inl.h

```diff
@@ -133,7 +133,7 @@ inline void url::update_base_search(std::string_view input,
 }
 
 inline void url::update_base_search(std::optional<std::string> input) {
```

Comment:
Instead of doing std::move all over, I recommend just doing this:

```suggestion
inline void url::update_base_search(std::optional<std::string>&& input) {
```

and then you can trim out at least one std::move... and you'll be helping the compiler by making the intention clear.